4.1.0 B2 Build Fail

### What is the issue?

On running "make" the process failed after about 10 minutes

```
[ 42%] Building CXX object libtransmission/CMakeFiles/transmission.dir/magnet-metainfo.cc.o
[ 42%] Building CXX object libtransmission/CMakeFiles/transmission.dir/makemeta.cc.o
[ 43%] Building CXX object libtransmission/CMakeFiles/transmission.dir/net.cc.o

/root/transmission-4.1.0.B2/libtransmission/net.cc: In function ‘tr_socket_t {anonymous}::tr_netBindTCPImpl(const tr_address&, tr_port, bool, int*)’:
/root/transmission-4.1.0.B2/libtransmission/net.cc:329:27: error: ‘evutil_make_listen_socket_ipv6only’ was not declared in this scope
     if (addr.is_ipv6() && evutil_make_listen_socket_ipv6only(fd) != 0 &&
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/transmission-4.1.0.B2/libtransmission/net.cc:329:27: note: suggested alternative: ‘evutil_make_listen_socket_reuseable’
     if (addr.is_ipv6() && evutil_make_listen_socket_ipv6only(fd) != 0 &&
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                           evutil_make_listen_socket_reuseable
make[2]: *** [libtransmission/CMakeFiles/transmission.dir/build.make:426: libtransmission/CMakeFiles/transmission.dir/net.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:601: libtransmission/CMakeFiles/transmission.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```
None of it means much to me, any ideas on whats wrong ?

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

4.1.0 Beta 2

--- a/libtransmission/net.cc
+++ b/libtransmission/net.cc
@@ -326,13 +326,17 @@ tr_socket_t tr_netBindTCPImpl(tr_address const& addr, tr_port port, bool suppres
     (void)setsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, reinterpret_cast<char const*>(&optval), sizeof(optval));
     (void)evutil_make_listen_socket_reuseable(fd);
 
-    if (addr.is_ipv6() && evutil_make_listen_socket_ipv6only(fd) != 0 &&
+#ifdef IPV6_V6ONLY
+    // TODO(tearfur): Consider using `evutil_make_listen_socket_ipv6only` once minimum libevent version is bumped to 2.1.9 or above
+    if (addr.is_ipv6() &&
+        setsockopt(fd, IPPROTO_IPV6, IPV6_V6ONLY, reinterpret_cast<char const*>(&optval), sizeof(optval)) == -1 &&
         sockerrno != ENOPROTOOPT) // if the kernel doesn't support it, ignore it
     {
         *err_out = sockerrno;
         tr_net_close_socket(fd);
         return TR_BAD_SOCKET;
     }
+#endif
 
     auto const [sock, addrlen] = tr_socket_address::to_sockaddr(addr, port);
 

